### PR TITLE
Allow attr version in setup.cfg

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: .github/workflows/install_test.yml [hunk 1]
```diff
@@ -8,12 +8,13 @@ on:
 name: Install Test
 jobs:
   install_ww_complete:
-    name: ${{ matrix.python_version }} install woodwork complete
-    runs-on: ubuntu-latest
+    name: ${{ matrix.python_version }} - ${{ matrix.python_version }} install woodwork complete
```
[the remark this answers is not in the document]
Comment (gsheni) [reply .github/workflows/install_test.yml]: yep. fixed

File: Makefile [hunk 2]
```diff
@@ -28,9 +28,7 @@ testcoverage:
 	pytest woodwork/ --cov=woodwork
 
 .PHONY: installdeps
-installdeps:
-	pip install --upgrade pip
-	pip install -e .
+installdeps: upgradepip
```
Comment (jeff-hernandez) [inline Makefile:31]: Do we still need `installdeps-dev`?

File: Makefile [hunk 3]
```diff
@@ -28,9 +28,7 @@ testcoverage:
 	pytest woodwork/ --cov=woodwork
 
 .PHONY: installdeps
-installdeps:
-	pip install --upgrade pip
-	pip install -e .
```
Comment (ParthivNaresh) [inline Makefile:33]: Why was this removed? Shouldn't we expand this to cover
`pip install -e .`, `pip install -e ".[dev]"` and `pip install -e ".[test]"`

File: Makefile (hunk 2 above)
Comment (gsheni) [reply Makefile:31]: removed

File: Makefile (hunk 3 above)
Comment (gsheni) [reply Makefile:33]: Additionally, we don't use the test, and dev commands anywhere (CI wise). Devs can just directly install them via `pip`.

**Review by jeff-hernandez — APPROVED** [review]
